sources/packages/backend/apps/test-db-seeding/src/seed-executors/seed-executor.ts

```diff
@@ -0,0 +1,109 @@
+import {
+  DiscoveredClassWithMeta,
+  DiscoveryService,
+} from "@golevelup/nestjs-discovery";
+import { Injectable, OnModuleInit } from "@nestjs/common";
+import { DATA_SEED, DATA_SEED_METHOD } from "./data-seed.decorator";
+
+@Injectable()
+export class SeedExecutor implements OnModuleInit {
+  constructor(private readonly discoveryService: DiscoveryService) {}
+
+  async onModuleInit(testServicesClasses?: string[]): Promise<void> {
+    // Discovers all providers in the test-db-seeding app that have meta at a specific
+    // key and returns the provider(s) and associated meta.
+    const discoveredClasses =
+      await this.discoveryService.providersWithMetaAtKey<{
+        name: string;
+        order: number;
+        skip: boolean;
+      }>(DATA_SEED);
+
+    // Remove skipped classes.
+    let classesToBeSeeded = discoveredClasses.filter(
```

Comment:
I would suggest that we have a discovery method that returns a single object array. This object can have all possible metadata information needed to apply the filter now or in the future. Any filter or sort to be applied should be as simple as the one on the like 23.
The object can hold the information like:
```
DataSeed metadata
DataSeedMethod metadata
method handlers for execution
```
This class can have one private method to do the discovery and prepare the array of objects. Then the executed can apply the filters and the sort.